src/components/NewDatePicker/index.js

```diff
@@ -47,11 +47,12 @@ class NewDatePicker extends React.Component {
 
     /**
      * Trigger the `onInputChange` handler when the user input has a complete date or is cleared
-     * @param {Date} selectedDate
+     * @param {string} selectedDate
      */
     setDate(selectedDate) {
         this.setState({selectedDate}, () => {
             this.props.onInputChange(selectedDate);
+            this.props.onBlur(); // force validation on every date change
```

Comment:
This is a weird or sort of hacky thing to do. This will make validation occur twice.

The problem is in the `Form` Component. Specifically here

https://github.com/Expensify/App/blob/756e9d82787f8a7c1587857c2310bd807e913f09/src/components/Form.js#L196

The `touchedInputs` is not set.

We set `touchedInputs` here on blur. This makes sense as we don't want to show an error on every change in the input, instead only when focusing on a different input in the form.

https://github.com/Expensify/App/blob/756e9d82787f8a7c1587857c2310bd807e913f09/src/components/Form.js#L296-L303

But in our case validation needs in every change maybe we should look for a way to set the `touchedInputs` once or prop to just to show validation errors without considering the `touchedInputs`

your thoughts? @burczu @roryabraham

Reply:
I wouldn't say it's hack-ish solution here - custom form fields like this date picker should decide on their own when they are considered "blurred", and call suitable event handler then.

Of course, it is also connected to the touched state, but I think this state is more internal thing from the form perspective. What I think as an alternative solution is to provide the `onTouched` event handler, that could be called by the form field instead of calling `onBlur` but effectively it would be the same thing done in slightly different way.

Reply:
> What I think as an alternative solution is to provide the onTouched event handler, that could be called by the form field instead of calling onBlur but effectively it would be the same thing done in slightly different way.

@burczu Can you elaborate more with implementation details?

@roryabraham Your thoughts so far?

Reply:
@Santhosh-Sellavel Here, in the `Form.js`:

https://github.com/Expensify/App/blob/dd5fbb6d12e5daa66458b3db07f0ea88dbc4d449/src/components/Form.js#L285-L329

we can inject additional prop `onTouched` to the children:

```javascript
onBlur: () => {
    ...
},
onTouched: () => {
    this.setTouchedInput(inputID);
},
onInputChange: () => {
    ...
}
```

and then, in new the `NewDatePicker` component, instead of calling `this.props.onBlur()` ever time we change the date, we could call `this.props.onTouched`.

Effect is the same, but I don't think it is the best option - the `touchedInput` state is an internal value of the `Form` component and I don't we think we should allow fields to change it directly. 

Also, when you check to docs about [Creating and using Forms](https://github.com/Expensify/App/blob/main/contributingGuides/FORMS.md) you can find [here](https://github.com/Expensify/App/blob/main/contributingGuides/FORMS.md#props-provided-to-form-inputs) that the `onBlur` callback is designed to call validation directly (at least I understand it this way):

"onBlur: An onBlur handler that calls validate."

As I wrote in my previous comment - I believe the custom field should decide when it is considered as blurred, so calling `this.props.onBlur` on every date change might be correct.

Reply:
Looks good to me your thoughts @roryabraham

Reply:
yes, @roryabraham - your opinion would be appreciated here

Reply:
@roryabraham bump!

Reply:
> I believe the custom field should decide when it is considered as blurred, so calling this.props.onBlur on every date change might be correct

The only thing I can imagine might be problematic with this is that it could cause the input to lose accessibility focus with every change, which would be annoying for screen readers. However, I'm not confident in this because I'm not experienced with using screen readers

> the touchedInput state is an internal value of the Form component and I don't believe we should allow form fields to amend it directly

I don't think I agree with this. If a child input should be able to control when it is blurred, shouldn't it also be able to control when it's touched?

---

So I'd suggest the alternate solution where we give the child an `onTouched` prop.

Reply:
@Santhosh-Sellavel @roryabraham the alternate solution is now applied